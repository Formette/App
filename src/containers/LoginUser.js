// @flow
import React from 'react';
import { graphql, compose } from 'react-apollo'
//Components
import AuthLayout from '../components/organisms/AuthLayout';
import {Input, Button, Link} from '../components/atoms/index';
import Error from '../components/molecules/Error';
//Styles
import Colors from '../styles/Colors';
//API
import {SIGIN_USER_MUTATION} from '../api/Mutations';
//Utilities
import {_saveUsername, _saveUserId} from '../services/utilities';

class LoginUser extends React.PureComponent {
    props: {
        signinUser: any,
        router: any,
    };
    state = {
        email: "",
        password: "",
        error: false,
        errorMsg: ""
    };
    _onSignIn = _ => {
      const {email, password} = this.state;
        //Verifies if the inputs are empty or not
        if(email && password) {
            //logs in the user
            this.props.signinUser({variables: {email, password}})
                .then((res) => {
                    window.localStorage.setItem('graphcoolToken', res.data.signinUser.token);
                    _saveUsername(res.data.signinUser.user.userName);
                    _saveUserId(res.data.signinUser.user.id);
                }).then(_ => this.props.router.replace('/'))
                .catch((e) => {
                    console.log(e);
                    this.setState({error: true, errorMsg: "Ops! Invalid Email or password."});
                })
        }else{
            this.setState({
                error: true,
                errorMsg: "This form is feeling lonely, needs affection, needs data."
            });
        }
    };
    render(){
        const {email, password, error, errorMsg} = this.state;
        return(
           <AuthLayout description="Welcome back, Come quick! Your forms are waiting for you">
               <label htmlFor="signinEmail" className="sr-only">Email address</label>
                   <Input id="signinEmail"
                          type="email"
                          value={email}
                          onChange={(e) => this.setState({email: e.target.value})}
                          className="form-control"
                          placeholder="Email address"
                          required autoFocus/>

               <label htmlFor="signinPassword" className="sr-only">Password</label>
                   <Input id="signinPassword"
                          type="password"
                          value={password}
                          onChange={(e) => this.setState({password: e.target.value})}
                          className="form-control"
                          placeholder="Password"
                          required autoFocus/>

               <Button className="btn btn-lg btn-primary btn-block"
                       onClick={this._onSignIn}
                       style={{marginTop: 10, marginBottom: 10}}
                       color={Colors.primary}>Sign In</Button>

               <Link href="/signup">Do not have an account yet? Omg is free. <u><strong>Create here!</strong></u></Link>
               <Error show={error}>{errorMsg}</Error>
           </AuthLayout>
        )
    }
}

const LoginUserWithData = compose(
    graphql(SIGIN_USER_MUTATION, {name: 'signinUser'}),
);

export default LoginUserWithData(LoginUser);


