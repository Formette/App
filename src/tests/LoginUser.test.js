// LoginUser.test.js
import React from "react";
import LoginUser from "../containers/LoginUser";
import renderer from "react-test-renderer";

describe("Login", () => {
  it("UI renders correctly", () => {
    const tree = renderer.create(<LoginUser />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
