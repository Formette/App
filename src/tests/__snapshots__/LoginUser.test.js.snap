// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login UI renders correctly 1`] = `
<div
  className="container sc-gZMcBi dCdvRp"
>
  <form
    className="form-signin"
  >
    <h2
      className="sc-bxivhb kRbHZm"
    >
      Formette
    </h2>
    <h6
      className="sc-bZQynM hpyxrd"
    >
      Welcome back, Come quick! Your forms are waiting for you
    </h6>
    <label
      className="sr-only"
      htmlFor="signinEmail"
    >
      Email address
    </label>
    <input
      autoFocus={true}
      className="form-control sc-bwzfXH kUMhZx"
      id="signinEmail"
      onChange={[Function]}
      placeholder="Email address"
      required={true}
      type="email"
      value=""
    />
    <label
      className="sr-only"
      htmlFor="signinPassword"
    >
      Password
    </label>
    <input
      autoFocus={true}
      className="form-control sc-bwzfXH kUMhZx"
      id="signinPassword"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      className="btn btn-lg btn-primary btn-block sc-bdVaJa hayiWv"
      color="#7950f2"
      onClick={[Function]}
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
      type="button"
    >
      Sign In
    </button>
    <a
      className="sc-htpNat jfadgg"
      onClick={[Function]}
    >
      Do not have an account yet? Omg is free.
       
      <u>
        <strong>
          Create here!
        </strong>
      </u>
    </a>
  </form>
</div>
`;
